### ISSUE TITLE
ACL-2016-End-to-End Relation Extraction using LSTMs on Sequences and Tree Structures

### ISSUE THREAD

**Issue by BrambleXu** [BrambleXu/knowledge-graph-learning#194]
**一句话总结：**

可以抽取entity和relation的end-to-end模型, entity的建模通过Bi-LSTM，relation的建模通过tree-structured LSTM-RNN。

> 问题：
> 提案：RNN based的模型用于捕捉word sequence和dependency tree。 
> 具体做法：stacking bidirectional tree-structured LSTM-RNNs on bidirectional sequential LSTM-RNNs
> 效果：This allows our model to jointly represent both entities and relations with shared parameters in a single model.
> ? : We further encourage detection of entities during training and use of entity information in relation extraction via entity pretraining and scheduled sampling. 

**资源：**

- [pdf](https://aclweb.org/anthology/papers/P/P16/P16-1105/)
- [code](
- [paper-with-code](

**论文信息：**

- Author: Toyota Technological Institute
- Dataset: 
- keywords:  end-to-end

**笔记：**

之前NER和RE是按pipeline的方式来处理的，但是 recent studies show that end-to-end (joint) modeling of entity and relation is important for high performance (Li and Ji, 2014; Miwa and Sasaki, 2014) since relations interact closely with entity information. 
> 为什么二者有相乘效果，有具体的分析吗

There are two ways to represent relations between entities using neural networks: recurrent/recursive neural networks (RNNs) and convolutional neural networks (CNNs).
> 这个分类太粗糙了吧。这只是模型而不是方法

其中RNN能直接表达linguistic structures: word sequences (Hammerton, 2001) and constituent/dependency trees (Tai et al., 2015). 而对于RE任务，之前一些研究的报告显示RNN效果比CNN差。但是这些LSTM-based system大部分只包含 limited linguistic structure和neural architectures，and do not model entities and relation jointly. 

而这里为了提高RE的效果，使用的feature是Word sequence and tree structure. 

我们的end-to-end模型用于提取两个entity之间的word sequence和dependency tree structure来帮助提高RE的效果。用bidirectional sequential (left-to-right and right-to-left)来对entities建模，用bidirectional tree-structured (bottom-up and top-down) LSTMRNNs 对relations来建模。

我们的模型先检测出entities，然后根据检测的entity来推测relation，NN 参数通过entity和relation的task以joint的方式更新。

我们模型与传统end-to-end relation抽取模型的不同，我们的模型在训练上有两点提高：
-  entity pretraining, which pretrains the entity model, 
-  scheduled sampling (Bengio et al., 2015), which replaces (unreliable) predicted labels with gold labels in a certain robability. 

这些提高减缓了the problem of low-performance entity detection in early stages
of training, as well as allow entity information to further help downstream relation classification

**模型图：**

![image](https://user-images.githubusercontent.com/10768193/58138504-60e0ba80-7c71-11e9-8f33-02f849eb6b27.png)

**结果**：

**接下来要看的论文**：
